src/ContactSet.Search/ContactSet.Search.Api.Tests/Tests.ContactSetSearchFunc/Test.Invoke.cs

```diff
@@ -0,0 +1,160 @@
+﻿using DeepEqual.Syntax;
+using GGroupp.Infra;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GGroupp.Internal.Support.ContactSet.Search.Api.Tests;
+
+partial class ContactSetSearchFuncTest
+{
+
+    [Fact]
+    public void InvokeAsync_CancellationTokenIsCanceled_ExpectTaskIsCanceled()
+    {
+        var success = new DataverseSearchOut(0, null);
+        var mockDataverseApiClient = CreateMockDataverseApiClient(success);
+
+        var func = CreateFunc(mockDataverseApiClient.Object);
+
+        var valueTask = func.InvokeAsync(new(string.Empty, new()), new CancellationToken(true));
+        Assert.True(valueTask.IsCanceled);
+    }
+
+    [Theory]
+    [InlineData("\u043B\u044C\u0441")]
+    [InlineData(Strings.Empty)]
+    [InlineData(null)]
+    public async Task InvokeAsync_CancellationTokenIsNotCanceled_ExpectCallDataVerseApiClientOnce(
+        string searchString)
+    {
+        const string guidString = "2a5d892f-1400-ec11-94ef-000d3a4a099f";
+        var success = new DataverseSearchOut(0, null);
+        var mockDataverseApiClient = CreateMockDataverseApiClient(success, IsMatchDataverseInput);
+
+        var token = new CancellationToken(false);
+
+        var func = CreateFunc(mockDataverseApiClient.Object);
+        _ = await func.InvokeAsync(new(searchString, new(guidString)), token);
+
+        mockDataverseApiClient.Verify(
+            c => c.SearchAsync(
+                It.IsAny<DataverseSearchIn>(), token),
+            Times.Once);
+
+        void IsMatchDataverseInput(DataverseSearchIn actual)
+        {
+            var expected = new DataverseSearchIn(
+                $"*{searchString}*")
+            { 
+                Entities = new ReadOnlyCollection<string>(new[] { "contact" }), 
+                Filter = $"parentcustomerid eq '{Guid.Parse(guidString).ToString("D", CultureInfo.InvariantCulture)}'"
```

Comment:
Why do you use Guid.Parse(guidString).ToString("D", CultureInfo.InvariantCulture)? You have guidString